### PR TITLE
Popup + omnibox integration

### PR DESCRIPTION
**Pull request by vesln**
ext-omnibox

ext-search

## Popup

Initially, I started with the idea that this could be a react component as the rest of the system. Sadly, the iframes won't fire up the `load` event and I couldn't think of a way to work around this. The good news is that this piece is simple enough and we can easily transform it in the very near future. I've taken the loading indicator from the main PH app. 

The name of the micro app is "popup" instead of "search" since search is only the first feature we'd like to implement. If you have suggestions for better naming, please feel free to share them :)

## Omnibox

The simplest implementation possible. I'm hoping we can enhance this in the future and add collection search, people search and eventually suggestions. I think it's a good start and we can experiment with it.

## Structure

I've structured the files inside "micro apps", that are encapsulated but still can use shared components. We've got "popup", "background" and "tabs". While we can put them inside top level directories, this will make reusing (requires, imports) uglier, so I decided to nest their items under `js` and `styles`.

## Configurations

`.env` files and `envify`. Nuff said.

## Development

Nothing too fancy, gulp powers everything. It's exactly what you'd expect.

## Tests

The end to end tests are powered by mocha, chai, selenium. In a nutshell we load the chrome extension with a chrome flag and we hit its pages through the extension id. Firing up selenium with the chromedriver is a manual process, which I'll be looking to automate in the future, but not a huge priority for now.

We'll be using jest for the unit tests.

## Reusing scss

This is something we have to start thinking about. Maybe extract a micro scss framework with producthunt styles (neat-like mixins)?

### CODE REVIEW

**Comment by andreasklinger** [comment]
@vesln what do you think - shall we split scope and notes in the PR description - makes it easier to understand

**Comment by andreasklinger** [comment]
really nice work :+1:

File: Gulpfile.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+/**
+ * File patterns.
+ */
+
+var patterns = {
+  html: 'src/**/*.html',
+  img: 'src/img/*.png',
+  css: 'src/styles/*.scss',
+  json: 'src/**/*.json'
+};
```
Comment (andreasklinger) [inline Gulpfile.js]: i like the app structure idea

what about doing

```
src/apps/popup/main.html
src/apps/popup/main.scss
src/apps/popup/main.js

src/apps/background/main.js

src/apps/tab/main.html
src/apps/tab/main.scss
src/apps/tab/main.js

src/common/loader/loader.scss
src/common/loader/loader.png
src/common/misc/stuff.scss
```

Comment (vesln) [inline Gulpfile.js]: I'll experiment with that idea, thanks

**Comment by vesln** [comment]
@andreasklinger re notes and scope - point taken

**Pull request merged**